src/systems/drive_to_pose_controller/DriveToPoseController.cc

```diff
@@ -0,0 +1,252 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "DriveToPoseController.hh"
+
+#include <gz/common/Profiler.hh>
+
+#include <gz/math/Pose3.hh>
+#include <gz/msgs/pose_v.pb.h>
+#include <gz/msgs/twist.pb.h>
+#include <gz/msgs/convert/Pose.hh>
+
+#include <gz/plugin/Register.hh>
+
+#include <gz/sim/Model.hh>
+#include <gz/transport/Node.hh>
+#include <gz/transport/TopicUtils.hh>
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+//////////////////////////////////////////////////
+class gz::sim::systems::DriveToPoseControllerPrivate
+{
+  /// \brief Callback for pose command
+  /// \param[in] _msg Pose message
+  public: void OnCmdPose(const msgs::Pose_V &_msg);
+
+  /// \brief Callback for odometry message
+  /// \param[in] _msg Current odometry message
+  public: void OnCurrentPose(const msgs::Pose_V &_msg);
+
+  /// \brief Calculate velocity and publish a twist message
+  public: void CalculateVelocity();
+
+  /// \brief Normalize the angles between -pi to pi
+  /// \param[in] _angle Angle to normalize
+  public: void NormalizeAngle(double &_angle);
+
+  /// \brief Gazebo communication node
+  public: transport::Node node;
+
+  /// \brief Model interface
+  public: Model model{kNullEntity};
+
+  /// \brief Velocity message publisher.
+  public: transport::Node::Publisher velocityPublisher;
+
+  /// \brief Pose reached publisher
+  public: transport::Node::Publisher poseReachedPublisher;
+
+  /// \brief Current pose of the model
+  public: std::optional<math::Pose3d> currentPose;
+
+  /// \brief Target pose of the model
+  public: std::optional<math::Pose3d> targetPose;
+
+  /// \brief Proportional gain for angular velocity
+  public: double angularPGain{2.0};
+
+  /// \brief Proportional gain for linear velocity
+  public: double linearPGain{1.0};
+
+  /// \brief Allowable angular deviation
+  public: double angularDeviation{0.05};
+
+  /// \brief Allowable linear deviation
+  public: double linearDeviation{0.1};
+
+  /// \brief Mutex to protect current and target poses
+  public: std::mutex mutex;
+};
+
+//////////////////////////////////////////////////
+DriveToPoseController::DriveToPoseController()
+    : dataPtr(std::make_unique<DriveToPoseControllerPrivate>())
+{
+  // Do nothing
+}
+
+//////////////////////////////////////////////////
+void DriveToPoseController::Configure(
+    const Entity& _entity,
+    const std::shared_ptr<const sdf::Element>& _sdf,
+    EntityComponentManager& _ecm,
+    EventManager& /*_eventMgr*/)
+{
+  this->dataPtr->model = Model(_entity);
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    gzerr << "This plugin should be attached to a model entity. "
+          << "Initialization failed." << std::endl;
+    return;
+  }
+
+  // Get linear P gain if available
+  if (_sdf->HasElement("linear_p_gain"))
```

Comment:
maybe we should add here some `gzdbg` or `gzwarm` to show which are the values of the gains

Reply:
Sounds good. I am now printing all the parameter values through `gzdbg`